### PR TITLE
Set up RealmSwift device tests

### PR DESCRIPTION
**Pull request by segiddins**
Closes https://github.com/realm/realm-cocoa/issues/1603.

\c @jpsim

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -160,9 +160,9 @@ test_ios_devices() {
         fi
         exit 1
     fi
-    configuration="$1"
+    configuration="$2"
     for device in "${serial_numbers[@]}"; do
-        xcrealm "-scheme 'iOS Device Tests' -configuration $configuration -destination 'id=$device' test"
+        $1 "-scheme 'iOS Device Tests' -configuration $configuration -destination 'id=$device' test"
```
Comment (alazier) [inline build.sh]: could make another variable like `configuration` for clarity and consistency.

**Comment by alazier** [comment]
Other than my one comment :+1:

File: build.sh (hunk 1 above)
Comment (segiddins) [inline build.sh]: done

**Comment by jpsim** [comment]
:+1: last CI failure was unrelated.

**Pull request merged**